README.md

```diff
@@ -1,75 +1,138 @@
 
-```sh
-make docker-build docker-push IMG=<some-registry>/onload-operator:tag
-```
+### Build Onload images
 
-3. Deploy the controller to the cluster with the image specified by `IMG`:
+The required Onload images are: source, userland and kernel.
 
-```sh
-make deploy IMG=<some-registry>/onload-operator:tag
+1. Clone OpenOnload from https://github.com/Xilinx-CNS/onload
+2. Create an Onload source tarball:
 ```
-
-### Uninstall CRDs
-To delete the CRDs from the cluster:
-
-```sh
-make uninstall
+scripts/onload_mkdist
+```
+3. Create an Onload source image, to be pushed later to the image registry of your choice:
+```
+scripts/onload_mkcontainer --source kitchen-sink.kube.test:5000/onload-source:latest *.tgz
+```
+4. Push the Onload source image to the image registry:
+```
+docker push kitchen-sink.kube.test:5000/onload-source:latest
+```
+5. Create an Onload userland image:
+```
+scripts/onload_mkcontainer --user kitchen-sink.kube.test:5000/onload-user:latest *.tgz
+```
+6. Push the Onload userland image:
+```
+docker push kitchen-sink.kube.test:5000/onload-user:latest
 ```
+By default, the Onload userland uses UBI libc. Please check and patch the Dockerfile if this is incompatible with your application's environment.
 
-### Undeploy controller
-UnDeploy the controller from the cluster:
+7. Check out this branch of this repository.
+8. Build an Onload kernel image:
+```
+make -C build/onload-module/ onload-module ONLOAD_SOURCE_IMAGE_REPO=kitchen-sink.kube.test:5000/onload-source ONLOAD_SOURCE_IMAGE_TAG=latest ONLOAD_MODULE_IMAGE_REPO=kitchen-sink.kube.test:5000/onload-module
+```
+However, since we default to OpenShift, please also pass the correct `DTK_AUTO` and `KERNEL_VERSION` to build (and possibly cross-compile) the Onload and SFC kernel modules. I suggest examining Makefile.
 
-```sh
-make undeploy
+9. Push the Onload kernel image (I copied the autogenerated hash and kernel version):
+```
+docker push kitchen-sink.kube.test:5000/onload-module:deccdb8d036a4c7794dea2cda106b3c112b374a9-4.18.0-372.49.1.el8_6.x86_64
 ```
 
-## Contributing
-// TODO(user): Add detailed information on how you would like others to contribute to this project
 
-### How it works
-This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+### Prepare Kubernetes cluster
 
-It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/),
-which provide a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
+The Onload Operator v3 depends on third-party software:
 
-### Test It Out
-1. Install the CRDs into the cluster:
+1. Kernel Module Manamagent (KMM) Operator v1.1.1.
+2. CNI that supports ipvlan or macvlan configurations. The Onload Operator currently suggests Multus CNI. A sample configuration is https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/examples/macvlan-pod.yml.
```

Comment:
I don't think we are actually tied to `ipvlan` or `macvlan`. We should be able to use multus in general (although we have only tested with ipvlan/macvlan)
```suggestion
2. Multus CNI. A sample configuration using macvlan is https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/examples/macvlan-pod.yml.
```

Reply:
Thanks, applied. Also, added a note that `NetworkAttachmentDefinition` is used later.